tools/editor/scene_tree_dock.cpp

```diff
@@ -1774,9 +1784,22 @@ SceneTreeDock::SceneTreeDock(EditorNode *p_editor,Node *p_scene_root,EditorSelec
 	HBoxContainer *filter_hbc = memnew( HBoxContainer );
 	ToolButton *tb;
 
+	ED_SHORTCUT("scene_tree_dock/new_node", TTR("New Node"), KEY_MASK_CMD|KEY_A);
+	ED_SHORTCUT("scene_tree_dock/duplicate", TTR("Duplicate"), KEY_MASK_CMD|KEY_D);
+	ED_SHORTCUT("scene_tree_dock/move_up", TTR("Move Up"), KEY_MASK_CMD|KEY_UP);
+	ED_SHORTCUT("scene_tree_dock/move_down", TTR("Move Down"), KEY_MASK_CMD|KEY_DOWN);
+	ED_SHORTCUT("scene_tree_dock/delete_no_confirm", TTR("Delete (No Confirm)"), KEY_MASK_SHIFT|KEY_DELETE);
+	ED_SHORTCUT("scene_tree_dock/delete", TTR("Delete"), KEY_DELETE);
+	ED_SHORTCUT("scene_tree_dock/instance", TTR("Inherit Scene"));
+	ED_SHORTCUT("scene_tree_dock/replace", TTR("Change Type"));
+	ED_SHORTCUT("scene_tree_dock/add_script", TTR("Add Script"));
+	ED_SHORTCUT("scene_tree_dock/reparent", TTR("Reparent"));
+	ED_SHORTCUT("scene_tree_dock/merge_from_scene", TTR("Merge From Scene"));
+	ED_SHORTCUT("scene_tree_dock/save_branch_as_scene", TTR("Save Branch as Scene"));
+
 	tb = memnew( ToolButton );
 	tb->connect("pressed",this,"_tool_selected",make_binds(TOOL_NEW, false));
-	tb->set_tooltip(TTR("Add/Create a New Node")+"\n("+keycode_get_string(KEY_MASK_CMD|KEY_A)+")");
+	tb->set_tooltip(TTR("Add/Create a New Node")+"\n("+ED_GET_SHORTCUT("scene_tree_dock/new_node")->get_as_text()+")"); // cannot use $sc as we don't have the shortcut registered to the button
```

Comment:
IMO It would be better to fix the bug that prevents him from moving the code away from `_unhandled_key_input`. @Paulb23 Could you explain a bit more that bug?